pkg/controller/populators/import-populator.go

```diff
@@ -97,9 +100,29 @@ func NewImportPopulator(
 		return nil, err
 	}
 
+	if err := addEventWatcher(mgr, importPopulator); err != nil {
+		return nil, err
+	}
+
 	return importPopulator, nil
 }
 
+func addEventWatcher(mgr manager.Manager, c controller.Controller) error {
+	if err := c.Watch(source.Kind(mgr.GetCache(), &corev1.Event{}, handler.TypedEnqueueRequestsFromMapFunc[*corev1.Event](
+		func(_ context.Context, e *corev1.Event) []reconcile.Request {
+			if e.InvolvedObject.Kind == "PersistentVolumeClaim" && strings.Contains(e.InvolvedObject.Name, "prime") {
+				return []reconcile.Request{{
+					NamespacedName: types.NamespacedName{Name: e.Namespace},
```

Comment:
This looks weird, I was expecting you to create a key consisting of the Name and Namespace of the involved object. This looks like your name is the namespace of the event. Something like this should work
```go
NamespacedName: types.NamespacedName{Name: e.InvolvedObject.Name, Namespace: e.InvolvedObject.Namespace}
``